### PR TITLE
Add examples for `publish:bitbucketServer:pull-request` scaffolder action & improve related tests

### PR DESCRIPTION
**Pull request by parmar-abhinav**
## Hey, I just made a Pull Request!
Contribution of publish:bitbucketServer:pull-request examples as part of https://github.com/backstage/backstage/issues/18832

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by parmar-abhinav** [comment]
HI @benjdlambert, @tudi2d, @freben 

Can you please review this PR. 

I have raised some more PR's by adding examples to different actions. Can you please review the other PR's as well tagged with issue https://github.com/backstage/backstage/issues/18832

File: plugins/scaffolder-backend-module-bitbucket-server/src/actions/bitbucketServerPullRequest.examples.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        {
+          action: 'publish:bitbucketServer:pull-request',
+          id: 'publish-bitbucket-server-pull-request-target-branch',
+          name: 'Creating pull request on bitbucket server',
+          input: {
+            repoUrl: 'hosted.bitbucket.com?project=project&repo=repo',
+            title: 'My pull request',
+            sourceBranch: 'my-feature-branch',
+            targetBranch: 'development',
+            description: 'This is a detailed description of my pull request',
```
Comment (BillyMichael) [inline plugins/scaffolder-backend-module-bitbucket-server/src/actions/bitbucketServerPullRequest.examples.ts:72]: Should description be removed from this one as it isn't required for this example?

**Comment by BillyMichael** [comment]
@parmar-abhinav , it also looks like you have some `tsc` issues that need to be resolved before this can be merged.

File: plugins/scaffolder-backend-module-bitbucket-server/src/actions/bitbucketServerPullRequest.examples.ts (hunk 1 above)
Comment (parmar-abhinav) [reply plugins/scaffolder-backend-module-bitbucket-server/src/actions/bitbucketServerPullRequest.examples.ts]: Thanks for reviewing, removed description property. Please check

**Comment by parmar-abhinav** [comment]
@BillyMichael Completed requested changes please review

**Comment by parmar-abhinav** [comment]
Hi @BillyMichael

I am not able to understand why the GitHub actions are failing. Can you please help me in resolving these errors

**Comment by tudi2d** [comment]
Heads-up that the CI is failing due to type issues. Please run `yarn tsc` locally on your branch to figure out what type issues exist & how to fix them. We will review this PR as soon as the CI is green:)

**Comment by parmar-abhinav** [comment]
> Heads-up that the CI is failing due to type issues. Please run `yarn tsc` locally on your branch to figure out what type issues exist & how to fix them. We usually review PRs as soon as the CI is green:)

Thank you for your suggestion. Updated the PR, Please check.

**Comment by parmar-abhinav** [comment]
Hi @BillyMichael @tudi2d Can you please check the PR now and review it.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice thanks! :pray:

**Pull request merged**